kernel/pci/src/lib.rs

```diff
@@ -3,14 +3,17 @@
 //! Note: while pci currently uses port-io on x86 and mmio on aarch64,
 //! x86 may also support memory-based PCI configuration in the future;
 //! port-io is the legacy way to access the config space.
+//!
+//! For context on the various interrupt mechanisms (MSI/MSI-X/INTx):
+//! https://electronics.stackexchange.com/a/343218
```

Comment:
rustdoc wants you to surround raw links with `<` `>`
```suggestion
//! <https://electronics.stackexchange.com/a/343218>
```

Also, while this link is nice, it's not really anything official. Is there a more official source, such as a chapter/section of the PCI spec or an intel/arm technical reference manual?